packages/trace-viewer/src/ui/uiModeView.tsx

```diff
@@ -48,8 +48,8 @@ const xtermDataSource: XtermDataSource = {
 
 const searchParams = new URLSearchParams(window.location.search);
 const guid = searchParams.get('ws');
-const wsURL = new URL(`../${guid}`, window.location.toString());
-wsURL.protocol = (window.location.protocol === 'https:' ? 'wss:' : 'ws:');
+const wsURL = new URL(`/${guid}`, searchParams.get('server') ?? window.location.toString());
```

Comment:
In the common case, this will use the URL from `?server` as the base URL. That URL doesn't have any path anyways, so `../${guid}` will resolve to the same as `/${guid}`. As far as I know, codespaces doesn't perform any URL manipulation, but instead forwards based on ports, so this should be fine. Let me test that though.

Reply:
ToT is broken in codespaces. If you run `npx playwright test --ui --ui-host=localhost`, codespaces will open up `https://fantastic-space-goggles-x5p9q4v5v4g36xxx-45079.app.github.dev/trace/uiMode.html?ws=...&server=http://localhost:45079`. It's the same path as outside of codespaces, but a different hostname. It loads all the static assets fine, but the `server` parameter continues pointing to `localhost`, which it cannot access.

Reply:
this got fixed in https://github.com/microsoft/playwright/pull/33664